tests/JobDefinitionFactoryTest.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Tests;
+
+use App\JobDefinitionFactory;
+use Keboola\DockerBundle\Docker\Component;
+use Keboola\JobQueueInternalClient\JobFactory;
+use Keboola\JobQueueInternalClient\JobFactory\Job;
+use Keboola\ObjectEncryptor\ObjectEncryptor;
+use Keboola\ObjectEncryptor\ObjectEncryptorFactory;
+use Keboola\StorageApi\Client;
+use PHPUnit\Framework\TestCase;
+
+class JobDefinitionFactoryTest extends TestCase
+{
+    public function testCreateJobDefinitionWithConfigData(): void
+    {
+        $configData = [
+            'runtime' => [
+                'foo' => 'bar',
+            ],
+        ];
+
+        $jobData = [
+            'status' => JobFactory::STATUS_CREATED,
+            'projectId' => 'my-project',
+            'componentId' => 'my-component',
+            'configId' => 'my-config',
+            'configData' => $configData,
+        ];
+
+        $jobDefinitions = $this->createJobDefinitions($jobData, $configData);
+
+        self::assertCount(1, $jobDefinitions);
+        $jobDefinition = $jobDefinitions[0];
+
+        self::assertSame($jobData['configId'], $jobDefinition->getConfigId());
+        self::assertSame($jobData['componentId'], $jobDefinition->getComponentId());
+        self::assertSame('bar', $jobDefinition->getConfiguration()['runtime']['foo'] ?? null);
+    }
+
+    public function testCreateJobDefinitionWithConfigDataAndBackend(): void
+    {
+        $configData = [
+            'runtime' => [
+                'foo' => 'bar',
+                'backend' => [
+                    'type' => 'invalid',
+                ],
+            ],
+        ];
+
+        $jobData = [
+            'status' => JobFactory::STATUS_CREATED,
+            'projectId' => 'my-project',
+            'componentId' => 'my-component',
+            'configId' => 'my-config',
+            'configData' => $configData,
+            'backend' => [
+                'type' => 'custom',
+            ],
+        ];
+
+        $jobDefinitions = $this->createJobDefinitions($jobData, $configData);
+        self::assertSame(
+            $jobData['backend']['type'],
```

Comment:
```suggestion
            `custom`,
```

Reply:
Ok, ale proc? Vztahuje se to k tomu "neprogramovat v testech"? Ja si prave rikal, ze je z toho hezky videt, ze vysledkem musi byt proste stejna hodnota, ze to ukazuje ten vztah mezi vstupem a vystupem.

Reply:
"neprogramovat v testech" - asi trošku jo, jde o to, že ten test stavíš proti dynamické hodnotě, takže, dejme tomu, že upravíš `createJobDefinitions` tak, že bude $jobData předávat odkazem (což tady není nijak vidět) a vevnitř je přepíše - a najednou nevíš proti čemu to testuješ a klidně ten test může procházet úplně falešně. 
Když ti jde o tu vazbu tak je nejlepší z toho udělat konstantu a tu prdnout na všechny místa, kde ji chceš vidět.

Reply:
> bude $jobData předávat odkazem (což tady není nijak vidět) a vevnitř je přepíše

ty jo, to by me nenapadlo, ze by nekdo mohl chtit delat :D `createJobDefinitions` jsem bral jen jako loklani helper, protoze by se jinak to same opakovalo v kazdem test case.. navic se to jmenuje `createJobDefinition`, ne `updateJobData`, tak to nema co menit data atd atd.. ale chapu, ze smyslem testu je byt co nejexplicitnejsi, odolny proti zmenam

Reply:
No jasně, je to helper, ale ten to někam posílá a to něco s tím zase něco dělá a chápeš no. U toho pole je to asi trošku přitažený ze vlasy, ale kdyby to byl mutable objekt, tak už o tom nemůžeš říct vůbec nic.

Reply:
> kdyby to byl mutable objekt

Immutable FTW! ;) ale asi chapu tu myslenku.. nahradil jsem oba vyskyty